packages/account/src/Configs/personal-details-config.js

```diff
@@ -1,14 +1,13 @@
 import { toMoment, getErrorMessages, generateValidationFunction, getDefaultFields, validLength } from '@deriv/shared';
 import { localize } from '@deriv/translations';
 
-const personal_details_config = ({ residence_list, account_settings, is_appstore }) => {
+const personal_details_config = ({ residence_list, account_settings, is_appstore, getChangeableFields }) => {
     if (!residence_list || !account_settings) {
         return {};
     }
 
-    const disabled_items = [
-        ...Object.keys(account_settings).filter(field_name => field_name !== 'account_opening_reason' && !!field_name),
-    ];
+    const changeable_fields = getChangeableFields();
+    const disabled_items = Object.keys(account_settings).filter(field => !changeable_fields.includes(field));
```

Comment:
Nit: Since `changeable_fields` is used only in one place, may be u can replace it with the function itself